### ISSUE TITLE
Issues with typing on `defer`ed `resolve` method

### ISSUE THREAD

**Issue by shtaft** [elbywan/wretch#146]
Hey team!
Thank you so much for maintaining this package, we are absolutely loving it.

We're in the process of upgrading to V2, and have noticed a small Typescript issue during the upgrade. It appears that deferring a resolve chain is not supported in the V2 types.

Here is a small repro:

```ts
import wretch from 'wretch'

const handle403 = () => {
  console.log('Unauthorized!')
}

const client = wretch('https://my-base-url.com')
  .defer((wretch) => {
    return wretch
        // Argument of type '(wretch: Wretch<unknown, unknown, undefined>) => Wretch<unknown, unknown, WretchResponseChain<unknown, unknown, undefined>>'
        // is not assignable to parameter of type 'WretchDeferredCallback<unknown, unknown, undefined>'
        .resolve((_) => _.forbidden(handle403));
  })

})

```
```
Argument of type '(wretch: Wretch<unknown, unknown, undefined>) => 
    Wretch<unknown, unknown, WretchResponseChain<unknown, unknown, undefined>>'
is not assignable to parameter of type 'WretchDeferredCallback<unknown, unknown, undefined>'
```

We like to modify requests made in the deferred callback, but can't in V2. Are we doing something wrong?

Thank you so much!

**Comment by elbywan** [comment]
Hey @shtaft,

> Thank you so much for maintaining this package, we are absolutely loving it.

Thanks, this is heartwarming 🙇.

> We like to modify requests made in the deferred callback, but can't in V2. Are we doing something wrong?

No you're not 😉. There is definitely someting fishy with the return type of the `.defer()` function.

I just released a new version `2.1.4` including a fix which should hopefully solve the problem.

Feel free to reopen the issue otherwise, and thanks for reporting the bug 👍

**Comment by shtaft** [comment]
Unbelievable, thank you so much! Cannot believe your speed on resolving issues, incredible work.
